staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchPathProvider.java

```diff
@@ -109,6 +126,68 @@ public Path getTenantInprogressDirectory(final TenantId tenantId)
+                    .forEach(batch -> {
+                        try (Stream<Path> filePath = Files.walk(batch)) {
+                            size[0] = filePath.mapToLong(file -> file.toFile().length()).sum();
+                        } catch (IOException e) {
+                            LOGGER.info("Folder might have been deleted or moved.");
+                        }
+                        try (Stream<Path> files = Files.list(batch)) {
+                            Optional<Path> lastModifiedFile = files.max(Comparator.comparing(BatchPathProvider::getLastModified));
+                            if (lastModifiedFile.isPresent()) {
+                                Path file = lastModifiedFile.get();
+                                Map<String, Tracker> trackerMap = BatchProgressTracker.getInProgressTrackerMap();
+                                /*Check the file system and see differences between last modified time
+                                by making thread sleep for 1 second to confirm if upload is progressing*/
+                                Instant prevTime = getLastModified(file);
+                                threadSleep();
+                                Instant currTime = getLastModified(file);
+                                if (ChronoUnit.MILLIS.between(prevTime, currTime) > 0) {
+                                    Tracker tracker = new Tracker();
+                                    tracker.setLastModifiedTime(currTime);
+                                    tracker.setNumberOfBytesReceived(size[0]);
+                                    trackerMap.put(threadID, tracker);
+                                }
+                            }
+                        } catch (IllegalStateException | IOException e) {
+                            LOGGER.info("The file might have been moved to Completed.");
+                        }
+                    });
+        } catch (IOException e) {
+            LOGGER.info("The file might have been moved to Completed.");
+        }
+    }
+
+    private static Instant getLastModified(Path p)
+    {
+        try {
+            return Files.readAttributes(p, BasicFileAttributes.class).lastModifiedTime().toInstant();
+        } catch (IOException ioe) {
+            throw new IllegalStateException(ioe);
+        }
+    }
+
+    public List<String> getListOfInProgressThreadFromFileSystem(TenantId tenantId, BatchId batchId)
+    {
+        Path inProgressPath = getTenantInprogressDirectory(tenantId);
+        List<String> list = new ArrayList<>();
+        try (Stream<Path> stream = Files.list(inProgressPath)) {
+            stream.filter(batch -> BatchNameProvider.getBatchId(batch.getFileName().toString()).equals(batchId.getValue()))
+                    .forEach(path -> list.add(BatchNameProvider.extractThreadIDAndServiceID(path.getFileName().toString())));
+        } catch (IOException e) {
+            LOGGER.info("The folder might have been moved or deleted.");
```

Comment:
I don't understand this.  The in-progress folder might have been moved or deleted?  If that is the case then it sounds like an ERROR rather than an INFO.  Also, include the exception in the log if this is indeed an unexpected circumstance.  Have you ever seen this happening?

```suggestion
            LOGGER.error("The folder might have been moved or deleted.", e);
```

I presume you're intentionally returning an empty list in this circumstance?

Reply:
Yes, if at that exact moment when the batch completes, the in-progress folder moves to completed folder, and let's say we're checking the file system for the status, we might encounter IOException, so we are ignoring it as it's a natural exception bound to happen. And yes, returning the empty list in this circumstance.

Reply:
However, will change them to `error` level.

Reply:
I'm still concerned about this.  I see that the log message was changed in ed866afc8c8bc9ce8be5f7ffb17c696b11980397 and b1a4f1bf7f0e214a900f9c3e87bc49212d0f95bb and is now "Error while traversing In-progress folder for the tenant".  I'm going to mark the conversation as resolved and open a fresh conversation on it.